q2_diversity/tests/test_alpha_rarefaction.py

```diff
@@ -0,0 +1,428 @@
+            alpha_rarefaction(output_dir, t, max_depth=200, phylogeny=p)
+            index_fp = os.path.join(output_dir, 'index.html')
+            self.assertTrue(os.path.exists(index_fp))
+            self.assertTrue('observed_otus' in open(index_fp).read())
+            self.assertTrue('shannon' in open(index_fp).read())
+            self.assertTrue('faith_pd' in open(index_fp).read())
+
+    def test_alpha_rarefaction_with_phylogeny_and_metadata(self):
+        t = biom.Table(np.array([[100, 111, 113], [111, 111, 112]]),
+                       ['O1', 'O2'],
+                       ['S1', 'S2', 'S3'])
+        p = skbio.TreeNode.read(io.StringIO(
+            '((O1:0.25, O2:0.50):0.25, O3:0.75)root;'))
+        md = qiime2.Metadata(
+            pd.DataFrame({'pet': ['russ', 'milo', 'peanut']},
+                         index=['S1', 'S2', 'S3']))
+
+        with tempfile.TemporaryDirectory() as output_dir:
+            alpha_rarefaction(output_dir, t, max_depth=200, phylogeny=p,
+                              metadata=md)
+            index_fp = os.path.join(output_dir, 'index.html')
+            self.assertTrue(os.path.exists(index_fp))
+            self.assertTrue('observed_otus' in open(index_fp).read())
+            self.assertTrue('shannon' in open(index_fp).read())
+            self.assertTrue('faith_pd' in open(index_fp).read())
+
+    def test_invalid_invocations(self):
+        t = biom.Table(np.array([[100, 111, 113], [111, 111, 112]]),
+                       ['O1', 'O2'],
+                       ['S1', 'S2', 'S3'])
+        md = qiime2.Metadata(
+            pd.DataFrame({'pet': ['russ', 'milo', 'peanut']},
+                         index=['S1', 'S2', 'S3']))
+
+        with tempfile.TemporaryDirectory() as output_dir:
+
+            with self.assertRaisesRegex(ValueError, 'must be greater'):
+                alpha_rarefaction(output_dir, t, min_depth=200, max_depth=1,
+                                  metadata=md)
+
+            with self.assertRaisesRegex(ValueError, 'Provided steps'):
+                alpha_rarefaction(output_dir, t, max_depth=200, steps=1,
+                                  metadata=md)
+
+            with self.assertRaisesRegex(ValueError, 'Provided iterations'):
+                alpha_rarefaction(output_dir, t, max_depth=200, iterations=0,
+                                  metadata=md)
+
+            with self.assertRaisesRegex(ValueError, 'phylogeny was not'):
+                alpha_rarefaction(output_dir, t, max_depth=200,
+                                  metadata=md, metric='faith_pd')
+
+            with self.assertRaisesRegex(ValueError, 'Unknown metric: pole'):
+                alpha_rarefaction(output_dir, t, max_depth=200,
+                                  metadata=md, metric='pole-position')
+
+            with self.assertRaisesRegex(ValueError, 'max_depth'):
+                alpha_rarefaction(output_dir, t, max_depth=1000)
+
+
+class ComputeRarefactionDataTests(unittest.TestCase):
```

Comment:
Do the test cases in this class need a random seed set, to avoid stochastic differences that may make the tests fail on repeated runs?